### PR TITLE
MISUV-7134 - Add repayment supplement credit from excess payment due to charge reduction to credits on View and Change

### PR DESCRIPTION
**Pull request by JordHemsworth**

### CODE REVIEW

File: app/models/financialDetails/responses/ChargesResponse.scala [hunk 1]
```diff
@@ -46,6 +46,10 @@ case class ChargesResponse(balanceDetails: BalanceDetails,
         _.mainType
       )
 
+      val mainTransaction = financialDetails.find(_.transactionId.equals(document.transactionId)).flatMap(
```
Comment (avelx) [inline app/models/financialDetails/responses/ChargesResponse.scala:49]: Please add type definition for mainTransaction val, this easy up reading this code

File: app/models/financialDetails/Payment.scala [hunk 2]
```diff
@@ -30,7 +30,8 @@ case class Payment(reference: Option[String],
                    dueDate: Option[LocalDate],
                    documentDate: LocalDate,
                    transactionId: String,
-                   mainType: Option[String] = None)
+                   mainType: Option[String] = None,
```
Comment (avelx) [inline app/models/financialDetails/Payment.scala:33]: can we check if we can remove defaults for main and mainTransactions ?

**Review by avelx — DISMISSED** [review]
approved with comments

File: app/models/financialDetails/Payment.scala (hunk 2 above)
Comment (owen-gilfellon) [reply app/models/financialDetails/Payment.scala]: Removed

**Review by vinay1yv — APPROVED** [review]

**Pull request merged**